tidy3d/web/simulation_task.py

```diff
@@ -0,0 +1,370 @@
+"""
+Tidy3d webapi types
+"""
+import os.path
+import tempfile
+from datetime import datetime
+from typing import List, Optional
+
+from pydantic import Extra, Field, parse_obj_as
+from tidy3d import Simulation
+from tidy3d.version import __version__
+
+from .cache import FOLDER_CACHE
+from .http_management import http
+from .s3utils import download_file, upload_file, upload_string
+from .tidy3d_types import (
+    Queryable,
+    ResourceLifecycle,
+    Submittable,
+    T,
+    Tidy3DResource,
+)
+
+SIMULATION_JSON = "simulation.json"
+SIMULATION_HDF5 = "output/monitor_data.hdf5"
+RUNNING_INFO = "output/solver_progress.csv"
```

Comment:
let's add webapi call for get running info rather than access csv directly.
so we can keep improving the running info on webapi side without introducing trouble on user's python client upgrade since the client upgrade is out of our contorl

Reply:
@lei-flex. So the `SimulationTask` has method `get_running_info()` that tries to download the `RUNNING_INFO` file and parse the contents. Is this good enough? or do you want another function in `webapi.py` that for example gets passed the task_id, loads the SimulationTask, and then calls this `get_running_info` method?